repo/tuf_on_ci/status.py

```diff
@@ -206,6 +202,17 @@ def status(verbose: int, push: bool) -> None:
                 success = False
 
     if push:
-        _git(["push", "origin", event_name])
+        try:
+            _git(["push", "origin", event_name])
+        except subprocess.CalledProcessError as e:
+            # Figure out if this is an error caused by remote being ahead
+            # of local branch
+            msg = "Updates were rejected because the remote contains work that you do"
+            found = e.stdout.find(msg)
+            if found:
+                print("Updates on remote, will not push")
```

Comment:
I'm trying to think how this works in the event and I _think_ it is handling the important cases: if there are actual metadata changes that we did not push, then `success==False` because those metadata are now unsigned... So we won't end up with singing-event saying the signing event is finished.

The message could be less about git internal and more about "There are new changes in signing event: skipping metadata update" or something but that's a nitpick

Reply:
> So we won't end up with singing-event saying the signing event is finished even though we skipped the push.

Yes, my understanding is that those changes may actually have been reverted by the change on remote, so the workflow that would be triggered by that change would reconcile the state and then push that.